pkgs/build-support/dotnet/add-nuget-deps/fetch-deps.sh

```diff
@@ -1,13 +1,27 @@
-set -e
+# shellcheck shell=bash
+set -euo pipefail
+
+export PATH="@binPath@:$PATH"
+
+LOCKFILE_OUTPUT="$1"
 
 genericBuild
 
-(
-    echo -e "# This file was automatically generated by passthru.fetch-deps.\n# Please dont edit it manually, your changes might get overwritten!\n"
-    @nugetToNix@/bin/nuget-to-nix "${NUGET_PACKAGES%/}"
-) > deps.nix
+nuget-to-json "${NUGET_PACKAGES%/}" >deps.json
+
+if [[ "$LOCKFILE_OUTPUT" == *.nix ]]; then
+    trap 'rm deps.json' exit
+
+    (
+        echo "# This file was automatically generated by passthru.fetch-deps."
+        echo "# Please dont edit it manually, your changes might get overwritten!"
+        echo -e "# TODO: This format file is obsolete, consider migrating to JSON.\n"
+        nuget-to-nix --convert deps.json
+    ) >deps.nix
 
-@nixfmt@/bin/nixfmt deps.nix
+    mv deps.nix "$LOCKFILE_OUTPUT"
```

Comment:
Sorry I didn't spot this earlier.

Sorry, I missed this before:

This reverts #361579 - is that intended?

If it's not intended, we should add `nixfmt-rfc-style` to `binPath` and then do:

```suggestion
    nixfmt deps.nix
    mv deps.nix "$LOCKFILE_OUTPUT"
```

Unless `nuget-to-nix --convert` is guaranteed to already be nixfmt compliant? After a quick skim through the code it looks like it isn't.

Reply:
That does revert it. I'd leave it out since we only added it to appease the CI check, but it's better to avoid any unintended breaking changes.

Reply:
> it's better to avoid any unintended breaking changes.

I can get behind this argument for not doing `nixfmt`. Let's leave this thread unresolved for now for visibility, maybe @corngood will have an opinion?

Reply:
> I can get behind this argument for not `nixfmt`ing. Maybe leave this thread unresolved for now for visibility, maybe corngood will have an opinion?

Honestly, I'd rather just have full compatibility with the previous version of `fetch-deps` to avoid any possible issues.
Given that we'll remove nix-based lockfiles next release, it'll only be there for a short while.